lib/math/math.go

```diff
@@ -0,0 +1,176 @@
+
+		"acosh": newUnaryBuiltin("acosh", math.Acosh),
+		"asinh": newUnaryBuiltin("asinh", math.Asinh),
+		"atanh": newUnaryBuiltin("atanh", math.Atanh),
+		"cosh":  newUnaryBuiltin("cosh", math.Cosh),
+		"sinh":  newUnaryBuiltin("sinh", math.Sinh),
+		"tanh":  newUnaryBuiltin("tanh", math.Tanh),
+
+		"log": newBinaryFunction("log", log, unpackLogArgs),
+
+		"e":  starlark.Float(math.E),
+		"pi": starlark.Float(math.Pi),
+	},
+}
+
+// floatOrInt is an Unpacker that converts a Starlark int or float to Go's float64.
+type floatOrInt float64
+
+func (p *floatOrInt) Unpack(v starlark.Value) error {
+	switch v := v.(type) {
+	case starlark.Int:
+		*p = floatOrInt(v.Float())
+		return nil
+	case starlark.Float:
+		*p = floatOrInt(v)
+		return nil
+	}
+
+	return fmt.Errorf("got %s, want float or int", v.Type())
+}
+
+// newUnaryBuiltin wraps a unary floating-point Go function
+// as a Starlark built-in that accepts int or float arguments.
+func newUnaryBuiltin(name string, fn func(float64) float64) *starlark.Builtin {
+	return starlark.NewBuiltin(name, func(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		var x floatOrInt
+		if err := starlark.UnpackPositionalArgs(name, args, kwargs, 1, &x); err != nil {
+			return nil, err
+		}
+		return starlark.Float(fn(float64(x))), nil
+	})
+}
+
+// newBinaryFunction wraps a binary floating-point Go function
+// as a Starlark built-in that accepts int or float arguments.
+func newBinaryFunction(name string, fn func(float64, float64) float64, unpackAgsFn func(string, starlark.Tuple, []starlark.Tuple) (floatOrInt, floatOrInt, error)) *starlark.Builtin {
+	return starlark.NewBuiltin(name, func(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		x, y, err := unpackAgsFn(name, args, kwargs)
+		if err != nil {
+			return nil, err
+		}
+		return starlark.Float(fn(float64(x), float64(y))), nil
+	})
+}
+
+// unpackRegularArgs unpacks the arguments of a regular binary function.
+func unpackRegularArgs(name string, args starlark.Tuple, kwargs []starlark.Tuple) (floatOrInt, floatOrInt, error) {
+	var x, y floatOrInt
+	if err := starlark.UnpackPositionalArgs(name, args, kwargs, 2, &x, &y); err != nil {
+		return 0, 0, err
+	}
+	return x, y, nil
+}
+
+// unpackLogArgs unpacks the arguments of the log function.
+func unpackLogArgs(name string, args starlark.Tuple, kwargs []starlark.Tuple) (floatOrInt, floatOrInt, error) {
+	var (
+		x    floatOrInt
+		base = floatOrInt(math.E)
+	)
+	if err := starlark.UnpackArgs(name, args, kwargs, "x", &x, "base?", &base); err != nil {
+		return 0, 0, err
+	}
+	return x, base, nil
+}
+
+func degrees(x float64) float64 {
+	return 360 * x / (2 * math.Pi)
+}
+
+func radians(x float64) float64 {
+	return 2 * math.Pi * x / 360
+}
+
+func log(x float64, base float64) float64 {
+	num := math.Log(x)
+	if base == 1 {
+		if num < 0 {
```

Comment:
Python reports an error (division by zero) if base == 1, and so too should Starlark.